docs/DeveloperGuide.md

```diff
@@ -1,26 +1,237 @@
+   If asked, choose to Open as Project (not Open as File).
+6. Click OK to accept the default settings but do ensure that the selected version of Gradle JVM matches the JDK 
+   being used for the project.
 
-### Value proposition
+## 3. Design
+This section describes the different components of the application and how each component interact with
+each other to run the program. 
+
+### 3.1 Architecture
+![architecture](../diagrams/architecture.png)
+
+The Architecture Diagram above provides a high-level view of the design of NUSchedule. The app can be broken down into 
+5 different components:  
+1. Main: Initializes the other components and connects them with each other.
+2. UI: Manages the User Interface that the user interacts with.
+3. Logic: Interprets user commands.
+4. Storage: Reads data from and writes data to the hard disk.
+5. Model: Stores the data the app uses in memory.
+
+__How the architecture components interact with each other__  
+The sequence diagram below shows how each individual component interacts with each other when the user inputs a command.
+![sequence](../diagrams/ArchitectureSequence.png)
+
+### 3.2 UI
+![UI Class Diagram](diagrams/Ui.png)
+
+*Figure 3.2.1 Class diagram for UI component*
+
+ __API__:`UI.java`  
+The UI consists of various parts, e.g. `printGreetingMessage`, `printEventList`, `showError`, `printNumEvent`, etc.
+ 
+The `UI` component reads user input using readCommand() and executes user commands according to the Logic component.
+
+### 3.3 Storage
+__API__:`Storage.java`
+The `Storage` component can save the list of event data in .txt format and read it back.   
+### 3.4 Logic
+__API__:`Parser.java`  
+
+![Logic Class Diagram](diagrams/LogicClass.png)
+
+*Figure 3.4.1 Class diagram for Logic component*
+
+The `logic` component parses the user input and executes commands based on the given input.
+ 1.	Parser reads the user input and returns a command to be executed by `NuSchedule`.
+ 2.	The command execution affects the EventList (e.g. clearing the list).
+ 3.	The result passes back to the UI, which then displays relevant feedback to the user.  
+
+### 3.5 Model
+__API__:`EventList.java`   
+The `Model` component stores an ArrayList, events, that represents the current list of events.
+
+### 3.6 Location
+![Location class diagram](diagrams/Location.jpeg)
+*Figure 3.6.1 Class diagram for location component*
+
+__API:__ `LocationList.java` <br>
+
+The `LocationList` is made up of an ArrayList of Locations, which is a type of variable that stores different information 
+for different types of location (eg. `Building`, `Hostel`, etc.). However, do note that any place that is 
+not within the list of saved locations will be saved as type `OutOfNUS`, and would not be saved into the `location.txt` 
+file. The location will still be part of the location list before the app closes, and it will also be 
+saved as part of the Event information.
```

Comment:
Is the Location class suppose to be an abstract class? Based on the enumeration, each location is of a certain type (building, hostel, LT or OutOfNUS) and there are classes for each of them, it looks like Location is an abstract class. Also, I think there should be an association between the Location and the enumeration

![image](https://user-images.githubusercontent.com/8433829/97528558-0496d400-19e9-11eb-9b9d-5579241bd38f.png)